scripts/groot_355/groot-355.py

```diff
@@ -0,0 +1,149 @@
+        "top_p": 0.9,
+        "top_k": 0,
+        "stream": False,
+        "presence_penalty": 0,
+        "frequency_penalty": 1,
+    }
+    headers = {
+        "Authorization": f"Bearer {os.environ['PERPLEXITY_KEY']}",
+        "Content-Type": "application/json",
+    }
+
+    response = requests.request("POST", url, json=payload, headers=headers)
+    url = json.loads(response.text)["choices"][0]["message"]["content"]
+    return url
+
+
+print("starting groot-355.py")
+
+canadian_fcu = set()
+number_of_entries = 0
+regex_pattern = r"^(" + "|".join([re.escape(name) for name in canadian_fcu]) + r")"
+# gets information from a file that just have the values you are looking for
+with open(canadian_fcu_names, "r") as file:
+    canadian_fcu.update({line.strip() for line in file})
+query = Q()
+for name in canadian_fcu:
+    query |= Q(legal_name__iexact=name)
+filtered_usnic = Usnic.objects.filter(query).values(
+    "ncua",
+    "website",
+    "country",
+    "legal_name",
+    "rssd",
+    "lei",
+    "fdic_cert",
+    "ein",
+    "occ",
+    "thrift_hc",
+    "cusip",
+)
+# selects the two fields to check for duplicates
+brand_values = Brand.objects.values("tag", "name")
+existing_tags = {x["tag"] for x in brand_values}
+existing_names = {x["name"] for x in brand_values}
+for row in filtered_usnic:
+    if row["legal_name"] in existing_names:
+        continue
+    existing_names.add(row["legal_name"])
+    # getting the datetime django wants
+    naive_datetime = datetime.now()
+    aware_datetime = timezone.make_aware(naive_datetime)
+    # removes special characters exept letters/numbers and spaces since we want to replace spaces with _
+    tag_name = re.sub(r"[^A-Za-z0-9 ]+", "", row.get("legal_name").lower())
+    mapped_rows = {
+        "created": aware_datetime,
+        "modified": aware_datetime,
+        "ncua": row.get("ncua"),
+        "countries": row.get("country"),
+        "name": row.get("legal_name"),
+        "aliases": row.get("legal_name").lower(),
+        "tag": generate_tag(tag_name.lower().replace(" ", "_"), 0, existing_tags),
+        "rssd": row.get("rssd"),
+        "lei": row.get("lei"),
+        "fdic_cert": row.get("fdic_cert"),
+        "ein": row.get("ein"),
+        "occ": row.get("occ"),
+        "thrift_hc": row.get("thrift_hc"),
+        "cusip": row.get("cusip"),
+    }
+    # adds tags to existing ones to check there are no duplicates
+    existing_tags.add(mapped_rows["tag"])
+    # checks if there is a website url
+    websiteurl = row.get("website")
+    if websiteurl != None:
+        websiteurl = check_website_URL(websiteurl)
+        mapped_rows["website"] = websiteurl
+    Brand.objects.create(**mapped_rows)
+    brand_instance = Brand.objects.get(name=mapped_rows["name"])
+    commentary = Commentary.objects.create(display_on_website=True, brand=brand_instance)
+    fcu = InstitutionType.objects.get(name="Credit Union")
+    commentary.institution_type.set([fcu])
```

Comment:
This will work, but it's not the best placement for `InstitutionType.objects.get(name="Credit Union")` The issue is that it will call the database each time you make the query. It would be better to make the query before the for loop even begins so that it just stays in memory and the database only gets hit once.